### PR TITLE
feature: add couchbase query runner

### PR DESCRIPTION
**Pull request by AntonZarutsky**
## What type of PR is this? (check all applicable)

- [x] Feature

## Description
Connector for Couchbase 5x+
1. Supports N1QL queries Couchbase 5x+ (maybe lower versions, didn't checked)
2. Supports Analytics service Couchbase 5.5+. (https://docs.couchbase.com/server/6.0/analytics/introduction.html)

- Access is done via Couchbase Rest API.
- JSON response transformation is carried from MongoDB connector.

## Mobile & Desktop Screenshots/Recordings (if there are UI changes)

Screen Shot 2019-03-29 at 17 37 01

### CODE REVIEW

File: redash/query_runner/couchbase.py [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+        return True
+
+    @classmethod
+    def annotate_query(cls):
+        return False
+
+    def test_connection(self):
+        result = self.call_service(self.noop_query, '')
+
+    def get_buckets(self, query, nameParam):
```
Comment (arikfr) [inline redash/query_runner/couchbase.py]: `nameParam` should be `name_param` to be consistent with PEP-8/our naming convention.

File: redash/query_runner/couchbase.py [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+
+    def call_service(self, query, user):
+        try:
+            user = self.configuration.get("user")
+            password = self.configuration.get("password")
+            host = self.configuration.get("host")
+            port = self.configuration.get('port', 8095)
+            params = {'statement': query}
+
+            url = "http://%s:%s/query/service" % (host, port)
```
Comment (arikfr) [inline redash/query_runner/couchbase.py]: Does Couchbase support HTTPS? In such case, we should probably make the protocol (http) configurable.

File: redash/query_runner/couchbase.py (hunk 1 above)
Comment (AntonZarutsky) [reply redash/query_runner/couchbase.py]: hi @arikfr. thx for review. Fixed nameParams and merged updates from master - then got test failing and locally message `ImportError: No module named flask_talisman` on `docker-compose` up run.

File: redash/query_runner/couchbase.py (hunk 2 above)
Comment (AntonZarutsky) [reply redash/query_runner/couchbase.py]: extracted as parameter

**Comment by AntonZarutsky** [comment]
Hi @arikfr. Updated PR. Can you check if we can proceed further?

**Comment by arikfr** [comment]
Merged. Thanks!

**Pull request merged**